### PR TITLE
LFA-804 add email entry fields to Create CSO account screen

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

https://justice.gov.bc.ca/jira/browse/FLA-804
- Work mostly from Celina
- Added email and confirmation email fields to Create CSO screen
- fields only show up for BCSC accounts.
-
_NOTE: styling/layout is not yet done, this will be tackled in [LFO-805](https://justice.gov.bc.ca/jira/browse/FLA-805)_

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

yarn lint
yarn test

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js [hunk 1]
```diff
@@ -27,16 +30,56 @@ export default function CSOAccount({
   const [termsAccepted, acceptTerms] = useState(false);
   const [continueBtnEnabled, setContinueBtnEnabled] = useState(false);
   const [showLoader, setShowLoader] = useState(false);
+  const [emailInput, setEmailInput] = useState({ email: "", confEmail: "" });
+  const [emailInputErrors, setEmailInputErrors] = useState({
+    emailError: "",
+    confEmailError: "",
+  });
+  const isBCeID = isIdentityProviderBCeID();
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js:38]: maybe change the logic, when the applicationInfo does not have an email prompt the user for a valid email address

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.scss [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.ct-cso-account {
+  * {
+    outline: none !important;
+  }
+  .error {
+    color: red;
+  }
+}
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.scss:8]:
```suggestion
}

```

**Review by alexjoybc — CHANGES_REQUESTED** [review]
small change on the logic for requesting email

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js (hunk 1 above)
Comment (KDNeufeld) [reply src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.js]: Done.  Logic is now based on simply whether application.email has data or not.

**Comment by alexjoybc** [comment]
LFA ;-)

File: src/frontend/efiling-frontend/package.json [hunk 3]
```diff
@@ -29,7 +29,8 @@
     "react-scripts": "3.4.3",
     "reactstrap": "^8.9.0",
     "regenerator-runtime": "^0.13.7",
-    "shared-components": "https://github.com/bcgov/react-shared-components/releases/download/v1.2.1/shared-components-1.2.1.tgz"
+    "shared-components": "https://github.com/bcgov/react-shared-components/releases/download/v1.2.1/shared-components-1.2.1.tgz",
+    "validator": "^13.5.2"
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/package.json:33]: When adding a new library we should explain it in the PR, what do you think? and add a link to the repo + git stats

**Review by alexjoybc — APPROVED** [review]

**Pull request merged**